contracts/contracts/Auth.sol

```diff
@@ -0,0 +1,81 @@
+// contracts/Auth.sol
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+contract Auth {
+    address[] public signers;
+    uint8 minThreshold;
+     // every time the validator set is updated, the version is incremented.
+     // increasing version numbers ensure that updateSigners() signatures cannot be reused
+    uint256 nextVersion;
+    event RegisterSigners(uint256 version, address[] signers, uint8 minThreshold);
+
+    mapping(bytes32 => bool) public fulfilledrequests;
+
+
+    constructor(address[] memory _signers, uint8 _minThreshold) {
+        _updateSigners(_signers, _minThreshold);
+    }
+
+    function _updateSigners(address[] memory _signers, uint8 _minThreshold) internal {
+        require(_signers.length < 256, "too many signers");
+        require(_minThreshold > signers.length / 2, "min threshold is too low");
```

Comment:
I wonder if we should use `3f + 1`? For `signers.length` equal `2`, `_minThreshold` is just one. From @nano-o:
> n/2 only prevents this kind of split-brain situation if validators never lie. In a threat model where an adversary takes over some validators,  we need at least 3f+1 validators to tolerate f compromised validators. So at minimum 4 validators with a threshold of 3 out of 4 (two sets of 3 have at least 2 in common, so if one lies we're still okay)